### PR TITLE
[DEVOPS-1381] Added button to filter projects in board

### PR DESCRIPTION
**Pull request by AndreKrebs**

### CODE REVIEW

File: core/src/main/resources/static/scripts/polyfill-loader.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+(function () {
+
+    'use_strict';
```
Comment (objective-solutions-user) [inline core/src/main/resources/static/scripts/polyfill-loader.js]: MAJOR Refactor or remove this statement. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3AS905)

File: core/src/main/resources/static/scripts/polyfill-loader.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        js.src = src;
+        js.async = false;
+        if (onLoad) {
+            js.onload = function() {
+                onLoad();
+            };
+        }
+        document.body.appendChild(js);
+    }
+
```
Comment (objective-solutions-user) [inline core/src/main/resources/static/scripts/polyfill-loader.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3ATrailingWhitespace)

File: core/src/main/resources/static/scripts/polyfill-loader.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+(function () {
+
+    'use_strict';
+
+    if (!isPositionStickySupported()) {
+        loadScript('/static/scripts/3rd/stickybits.min.js');
+    }
+
```
Comment (objective-solutions-user) [inline core/src/main/resources/static/scripts/polyfill-loader.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3ATrailingWhitespace)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [polyfill-loader.js#](https://github.com/objective-solutions/taskboard/blob/213ca79a9699b67af6a49fc98d1a512a54a652d8/core/src/main/resources/static/scripts/polyfill-loader.js#): Add a new line at the end of this file. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3AMissingNewlineAtEndOfFile)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [polyfill-loader.js#](https://github.com/objective-solutions/taskboard/blob/a946e138c5559c4491de62f3fa08139372436444/core/src/main/resources/static/scripts/polyfill-loader.js#): Add a new line at the end of this file. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3AMissingNewlineAtEndOfFile)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [polyfill-loader.js#](https://github.com/objective-solutions/taskboard/blob/f67f32ea1a149fc37da9f5b48f70b634199c5e3e/core/src/main/resources/static/scripts/polyfill-loader.js#): Add a new line at the end of this file. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3AMissingNewlineAtEndOfFile)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [polyfill-loader.js#](https://github.com/objective-solutions/taskboard/blob/1faa5d1dc97b384075e26780a8c378345ab27e20/core/src/main/resources/static/scripts/polyfill-loader.js#): Add a new line at the end of this file. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3AMissingNewlineAtEndOfFile)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html [hunk 4]
```diff
@@ -68,11 +110,45 @@
...
+                    .icon("taskboard-icons:cancel")
+                    .onClick(this._revertFilter.bind(this))
+                    .build();
+
+                    this._setInitialFilter();
+                },
+
+                attached: function() {
+                    if (window.stickybits)
+                        stickybits('[sticky-selector]')
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: Maybe we can pass the element to stickybits. Using with element, we avoid the stickybits search by the dom and apply in other components.
Then we can change the "/* Remove sticky-selector attr from element if you have to remove sticky position */" to "remove stickybits on attached function".

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html [hunk 5]
```diff
@@ -68,11 +110,45 @@
...
+                },
+
                 toggleAjax: function () {
+                    _onCardFieldsFiltersTap
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: Is this being used?

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html [hunk 6]
```diff
@@ -83,6 +159,30 @@
                     else
                         this.$.toggleIcon.icon = 'chevron-right';
                 },
+
+                _onClickSaveFilter: function() {
+                    this.$['button-filter-project'].disabled = true;
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: Use ```this.$$('#button-filter-project').disabled``` instead.

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html [hunk 7]
```diff
@@ -83,6 +159,30 @@
                     else
                         this.$.toggleIcon.icon = 'chevron-right';
                 },
+
+                _onClickSaveFilter: function() {
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: _onClickApplyFilter

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html [hunk 8]
```diff
@@ -83,6 +159,30 @@
...
+                    this.$['button-filter-project'].disabled = true;
+                    this._setInitialFilter();
+                    this.fire('iron-signal', {name: 'update-card-field-filter'});
+                },
+
+                _setInitialFilter: function() {
+                    this.initialFilters = cloneArray(this.cardFieldFilters);
+                },
+
+                _revertFilter: function() {
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: ```_onClickRevertFilter```
Or you can change the "_onClickSaveFilter" to "_applyFilter".

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html [hunk 9]
```diff
@@ -83,6 +159,30 @@
...
+                    this.set('cardFieldFilters', []);
+                    this.set('cardFieldFilters', cloneArray(this.initialFilters));
+                    this._updateUserPreferencesFilter();
+                },
+
+                _updateUserPreferencesFilter: function() {
+                    forEachInArray(this.cardFieldFilters, function(cardFieldFilter, index) {
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: You can remove the "index" parameters from functions. They aren't being used.

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html [hunk 10]
```diff
@@ -83,6 +159,30 @@
...
+                        forEachInArray(cardFieldFilter.filterFieldsValues, function(filter, index) {
+                            userPreferences.setFilter(filter.group, filter.value, filter.selected);
+                        });
+                    }.bind(this));
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: You can remove the ".bind(this)". There is no "this" inside the funtions.
Also you can use arrow functions.
```
forEachInArray(this.cardFieldFilters, (cardFieldFilter) => {
    forEachInArray(cardFieldFilter.filterFieldsValues, (filter) => {
        userPreferences.setFilter(filter.group, filter.value, filter.selected);
    });
});
```

File: cardboard/src/main/resources/static/elements/taskboard/project-preferences-modal.html [hunk 11]
```diff
@@ -128,6 +128,8 @@
                 },
 
                 _onOpen(cardFieldFilters) {
+                    if(!cardFieldFilters || cardFieldFilters.length == 0)
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/project-preferences-modal.html]: ===

File: cardboard/src/main/resources/static/elements/taskboard/taskboard-home.html [hunk 12]
```diff
@@ -614,7 +615,7 @@
                 },
 
                 refreshDone: function(cardFieldFilters, conf) {
-                    if (!conf || !cardFieldFilters)
+                    if (!conf || !cardFieldFilters || cardFieldFilters.length == 0)
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/taskboard-home.html]: ===

File: core/src/main/resources/static/elements/taskboard/config-table.html [hunk 13]
```diff
@@ -43,13 +49,17 @@
...
+                box-shadow: none;
+            }
         </style>
 
         <paper-material>
             <paper-toolbar on-tap="collapseMenuTap">
                 <paper-material class="config-item flex layout horizontal" elevation=0>
                     <iron-icon id="toggleIcon" class="config-icon icon-small" icon="chevron-right"></iron-icon>
-                    <paper-material class="config-title" elevation=0>Board Features</paper-material>
+                    <paper-material class="config-title" elevation=0>Board Layout</paper-material>
```
Comment (jhonypg) [inline core/src/main/resources/static/elements/taskboard/config-table.html]: As you're editing this line, please add " on elevation. ``` elevation="0"```

File: core/src/main/resources/static/elements/taskboard/filter-field-value.html [hunk 14]
```diff
@@ -52,6 +56,15 @@
                 -ms-transform: rotate(135deg);
                 transform: rotate(135deg);
             }
+
+            .filterName {
```
Comment (jhonypg) [inline core/src/main/resources/static/elements/taskboard/filter-field-value.html]: ".filter-name".
Then change on HTML too.
CamelCase isn't so usual on HTML/CSS.

File: application/src/test/java/objective/taskboard/it/MenuFilters.java [hunk 15]
```diff
@@ -180,4 +187,26 @@ public MenuFilters toggleAllProjects() {
         return this;
     }
 
+    public MenuFilters clickButtonSubmitFilter() {
+        buttonFilterApply.click();
```
[the remark this answers is not in the document]
Comment (AndreKrebs) [reply application/src/test/java/objective/taskboard/it/MenuFilters.java]: ok

File: application/src/test/java/objective/taskboard/it/MenuFilters.java [hunk 16]
```diff
@@ -180,4 +187,26 @@ public MenuFilters toggleAllProjects() {
...
+        return this;
+    }
+
+    public MenuFilters clickButtonRevertFilter() {
+        buttonRevertFilter.click();
```
[the remark this answers is not in the document]
Comment (AndreKrebs) [reply application/src/test/java/objective/taskboard/it/MenuFilters.java]: ok

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.WebElement;
+
+public class BoardFilterIT extends AuthenticatedIntegrationTest {
```
[the remark this answers is not in the document]
Comment (AndreKrebs) [reply application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: ok

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filter.html [hunk 18]
```diff
@@ -43,17 +48,15 @@
...
         </paper-toolbar>
+
         <iron-collapse id="collapseFilter">
             <paper-material class="layout vertical" elevation="0">
                 <template is="dom-repeat" items="{{cardFieldFilter.filterFieldsValues}}" as="filterFieldValue">
-                    <filter-field-value
-                            filter-field-value="{{filterFieldValue}}"
-                            on-selected-changed="_onFilterFieldValueSelectChanged"
-                            ></filter-field-value>
+                    <filter-field-value filter-field-value="{{filterFieldValue}}" >
```
[the remark this answers is not in the document]
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/card-field-filter.html]: ok

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html (hunk 4 above)
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: I made the change and tested it.

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html (hunk 5 above)
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: Removed

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html (hunk 6 above)
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: ok

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html (hunk 7 above)
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: ok

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html (hunk 8 above)
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: ok

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html (hunk 9 above)
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: ok

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html (hunk 10 above)
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: ok

File: cardboard/src/main/resources/static/elements/taskboard/project-preferences-modal.html (hunk 11 above)
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/project-preferences-modal.html]: ok

File: cardboard/src/main/resources/static/elements/taskboard/taskboard-home.html (hunk 12 above)
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/taskboard-home.html]: ok

File: core/src/main/resources/static/elements/taskboard/config-table.html (hunk 13 above)
Comment (AndreKrebs) [reply core/src/main/resources/static/elements/taskboard/config-table.html]: ok

File: core/src/main/resources/static/elements/taskboard/filter-field-value.html (hunk 14 above)
Comment (AndreKrebs) [reply core/src/main/resources/static/elements/taskboard/filter-field-value.html]: ok

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [polyfill-loader.js#](https://github.com/objective-solutions/taskboard/blob/45059a8adeeb39b7d6c211b4d68c28ccb8ef43c8/core/src/main/resources/static/scripts/polyfill-loader.js#): Add a new line at the end of this file. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3AMissingNewlineAtEndOfFile)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [polyfill-loader.js#](https://github.com/objective-solutions/taskboard/blob/4e8e96f02dc95e6ccc2515f11b9e457e869281e3/core/src/main/resources/static/scripts/polyfill-loader.js#): Add a new line at the end of this file. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3AMissingNewlineAtEndOfFile)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html [hunk 19]
```diff
@@ -68,10 +110,43 @@
...
+                        value: () => { return {}; }
+                    },
+                    revertFilterButton: {
+                        type: Object,
+                        value: () => { return {}; }
                     }
                 },
 
+                ready: function() {
+                    this.applyButton = ButtonBuilder('Apply')
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: Adjust the indentation for these buttons.

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filter.html [hunk 20]
```diff
@@ -100,18 +93,16 @@
...
                 _setCheckAll: function() {
                     this.$$('#checkAll').checked = false;
                     var filterFieldValueSelected = findInArray(this.cardFieldFilter.filterFieldsValues, function(filterFieldValue) {
                         return filterFieldValue.selected;
                     });
+
                     if (filterFieldValueSelected)
                         this.$$('#checkAll').checked = true;
 
+                    this.fire('initial-filter');
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/card-field-filter.html]: Remove this line. You can add an observable on card-field-filters to setInitialFilter.

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html [hunk 21]
```diff
@@ -43,19 +49,55 @@
...
         </style>
 
         <paper-material>
-            <paper-toolbar on-tap="_onCardFieldsFiltersTap">
-                <paper-material class="config-item flex layout horizontal" elevation=0>
+            <paper-toolbar>
+                <paper-material on-tap="_onCardFieldsFiltersTap" class="config-item flex layout horizontal" elevation=0>
                     <iron-icon id="toggleIcon" class="config-icon icon-small" icon="chevron-right"></iron-icon>
-                    <paper-material class="config-title" elevation=0>Card Field Filters</paper-material>
+                    <paper-material class="config-title" elevation=0>Board Filter</paper-material>
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: Add " on elevation

File: core/src/main/resources/static/elements/taskboard/filter-field-value.html [hunk 22]
```diff
@@ -52,6 +56,15 @@
                 -ms-transform: rotate(135deg);
                 transform: rotate(135deg);
             }
+
+            .filter-name {
```
Comment (jhonypg) [inline core/src/main/resources/static/elements/taskboard/filter-field-value.html]: Just "color" property is used here.

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java [hunk 23]
```diff
@@ -0,0 +1,145 @@
...
+        // Test "issues"
+        mainPage.lane("Deployable")
+                .boardStep("QAing")
+                .assertIssueList(
+                    "TASKB-237"
+                );
+    }
+
+    @Test
+    public void whenCreateNewUser_changeFilterAndApplyRevertAndVerifyIfReverted() {
```
[the remark this answers is not in the document]
Comment (AndreKrebs) [reply application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: Ok

File: core/src/main/resources/static/elements/taskboard/filter-field-value.html (hunk 22 above)
Comment (AndreKrebs) [reply core/src/main/resources/static/elements/taskboard/filter-field-value.html]: Ok

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html (hunk 21 above)
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: ok

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html (hunk 19 above)
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: Ok

File: application/src/test/java/objective/taskboard/it/MenuFilters.java [hunk 24]
```diff
@@ -79,6 +90,18 @@ public TeamsPage openTeamsConfiguration() {
...
+    public MenuFilters clickOptionMenuFilter(String option) {
+            Optional<WebElement> optionFound = optionCardFilter.stream()
+                    .filter(opt -> opt.getText().equals(option))
+                    .findFirst();
+
+            if (optionFound.isPresent()) {
+                waitForClick(optionFound.get());
+            }
+
+            return this;
```
[the remark this answers is not in the document]
Comment (AndreKrebs) [reply application/src/test/java/objective/taskboard/it/MenuFilters.java]: Ok

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java [hunk 25]
```diff
@@ -0,0 +1,145 @@
...
+
+public class BoardFilterIT extends AbstractUIWithCoverageIntegrationTest {
+
+    @Test
+    public void allFilterFeatures() {
+        MainPage mainPage = LoginUtils
+                .doLoginAsNewUser(webDriver)
+                .selectProjects(TASKB)
+                .submitSelectedProjects();
+
```
[the remark this answers is not in the document]
Comment (AndreKrebs) [reply application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: Ok

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java [hunk 26]
```diff
@@ -0,0 +1,145 @@
...
+        // Test "revert"
+        mainPage
+            .openMenuFilters()
+            .openCardFieldFilters()
+            .clickOptionMenuFilter("Project")
+            .assertBoardFilterFieldsOptionsUnmarked(
+                     "PROJ1 | PROJECT 1",
```
[the remark this answers is not in the document]
Comment (AndreKrebs) [reply application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: Ok

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java [hunk 27]
```diff
@@ -0,0 +1,145 @@
...
+                    "PROJ2 | PROJECT 2",
+                    "PROJ3 | PROJECT 3",
+                    "PROJ5 | PROJECT 5",
+                    "PROJ6 | PROJECT 6",
+                    "PROJ8 | PROJECT 8",
+                    "TASKB | TASKB-SUB-PROJ")
+            .clickOptionMenuFilter("Project")
```
[the remark this answers is not in the document]
Comment (AndreKrebs) [reply application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: Ok

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html [hunk 28]
```diff
@@ -83,6 +158,35 @@
...
+
+                _updateUserPreferencesFilter: function() {
+                    forEachInArray(this.cardFieldFilters, (cardFieldFilter) => {
+                        forEachInArray(cardFieldFilter.filterFieldsValues, (filter) => {
+                            userPreferences.setFilter(filter.group, filter.value, filter.selected);
+                        });
+                    });
+                },
+
+                _observerCardFieldFilters: function() {
```
[the remark this answers is not in the document]
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: Method removed

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filter.html (hunk 20 above)
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/card-field-filter.html]: I removed and refactored the code

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java [hunk 29]
```diff
@@ -0,0 +1,112 @@
...
+                .assertBoardFilterFieldsUnmarkedOptions(
+                        "PROJ1 | PROJECT 1",
+                        "PROJ2 | PROJECT 2",
+                        "PROJ3 | PROJECT 3",
+                        "PROJ5 | PROJECT 5",
+                        "PROJ6 | PROJECT 6",
+                        "TASKB | TASKB-SUB-PROJ");
+
+        // Test "issues"
+        mainPage.lane("Deployable")
```
Comment (jhonypg) [inline application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: On // Test "apply", eu should test some project, or another filter, that changes the issues in some lane.

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html [hunk 30]
```diff
@@ -43,19 +49,57 @@
...
+
+        <style is="custom-property">
+            tb-button {
+                --button-style: {
+                    text-transform: none;
+                }
+            }
         </style>
 
+        <iron-signals on-iron-signal-update-board-filters="_updateBoardFilters"></iron-signals>
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: You can use _setInitialFilter without creating _updateBoardFilters.

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java (hunk 29 above)
Comment (AndreKrebs) [reply application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: Added test on Team filter

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html (hunk 30 above)
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: Ok

**Pull request merged**